crates/sdl-encoder/src/description.rs

```diff
@@ -0,0 +1,135 @@
+use std::fmt::{self, Display};
+
+#[derive(Debug, PartialEq, Clone)]
+/// Convenience enum to create a Description. Can be a `Top` level, a `Field`
+/// level or an `Input` level. The variants are distinguished by the way they
+/// get displayed, e.g. number of leading spaces.
+pub enum Description {
+    /// Top-level description.
+    Top {
+        /// Description.
+        source: Option<String>,
+    },
+    /// Field-level description.
+    /// This description gets additional leading spaces.
+    Field {
+        /// Description.
+        source: Option<String>,
+    },
+    /// Input-level description.
+    /// This description get an additional space at the end.
+    Input {
+        /// Description.
+        source: Option<String>,
+    },
+}
+
+impl Display for Description {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Description::Top { source } => {
+                if let Some(description) = source {
+                    if is_block_string_character(description) {
+                        writeln!(f, "\"\"\"\n{}\n\"\"\"", description)?
+                    } else {
+                        writeln!(f, "\"\"\"{}\"\"\"", description)?
+                    }
```

Comment:
This can be done even more properly, where line 35 actually creates a `StringCharacter` that would make description look `"like this"` instead of `"""like this"""`. I am keeping it as is to match what Studio has at the moment. This will be aligned directly in `apollo-rs` encoder.